pbeApp/pbexport.cpp

```
@@ -362,42 +442,66 @@ void PBWriter::prepFile()
-        header.SerializeToCodedStream(&encstrm);
+    	if (!fileexists) {
+			google::protobuf::io::CodedOutputStream encstrm(&encbuf);
+			header.SerializeToCodedStream(&encstrm);
+    	}
     }
     encbuf.finalize();
 
-    outpb.open(fname.str().c_str());
-    outpb.write(&encbuf.outbuf[0], encbuf.outbuf.size());
+    outpb.open(fname.str().c_str(), std::fstream::app);
+    if (!fileexists) { //if file exists do not write header
+    	outpb.write(&encbuf.outbuf[0], encbuf.outbuf.size());
+    }
 }
 
-PBWriter::PBWriter(DataReader& reader)
+PBWriter::PBWriter(DataReader& reader, stdString pv)
     :reader(reader)
 	,info(reader.getInfo())
 	,year(0)
+    ,name(pv)
 {
     samp = reader.get();
 }
 
 void PBWriter::write()
 {
+	typeChangeError = 0;
     while(samp) {
-        prepFile();
-        (*transcode)(*this);
-        bool ok = outpb.good();
-        outpb.close();
-        if(!ok) {
-            std::cerr<<"Error writing file\n";
-            break;
-        }
+    	try {
+			prepFile();
+			if (!samp) break;
+			(*transcode)(*this);
+    	} catch (std::exception& up) {
+    		if (std::strstr(up.what(),"Error in data header")) {
+    			//Error in the data header means a corrupted sample data.
+    			//It can happen in the prepFile or in the transcode. Either way the resolution is the same.
+    			//We try to move ahead. If it doesn't work, abort.
+    			std::cerr<<"ERROR: "<<name.c_str()<<": Corrupted header, continuing with the next sample.\n"<<up.what()<<"\n";
+    			samp = reader.next();
+    		} else {
+    			//tough luck
+    			outpb.close();
+    			throw std::runtime_error(up.what());
+    		}
+    	}
+    	bool ok = outpb.good();
+		outpb.close();
+		if(!ok) {
+			std::cerr<<"Error writing file\n";
+			break;
+		}
     }
 }
 
 int main(int argc, char *argv[])
 {
+	//comment this if you want to see the protobuf logs
+	google::protobuf::LogSilencer *silencer = new google::protobuf::LogSilencer();
```

I assume this is to hide the errors from skipForward()?